main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1345,6 +1352,51 @@ object Lowering {
     Expression.Tuple(exps, tpe, pur, eff, loc)
   }
 
+  /**
+    * Returns a new `VarSym` for use in a let-binding.
+    *
+    * This function is called `mkLetSym` to avoid confusion with [[mkVarSym]].
+    */
+  private def mkLetSym(prefix: String, loc: SourceLocation)(implicit flix: Flix): Symbol.VarSym = {
+    val name = prefix + Flix.Delimiter + flix.genSym.freshId()
+    Symbol.freshVarSym(name, BoundBy.Let, loc)
+  }
+
+  /**
+    * Returns a tuple expression that is evaluated in parallel.
+    */
+  private def mkParTuple(tuple: Expression.Tuple)(implicit flix: Flix): Expression = {
```

Comment:
It would be nice to have a comment that details the transformation. E.g. `par (a, b, c)` transforms into `[the lines that are generated]`

Reply:
And maybe an inline comment for each chunk of expressions you build